sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/PartitionReceiverTests.cs

```diff
@@ -638,5 +948,66 @@ public async Task DisposeAsyncDelegatesToCloseAsync()
                 typeof(PartitionReceiver)
                     .GetProperty("RetryPolicy", BindingFlags.Instance | BindingFlags.NonPublic)
                     .GetValue(partitionReceiver);
+
+        /// <summary>
+        ///   Allows the observation of the creation of a <see cref="TransportConsumer"/>
+        ///   for testing purposes.
+        /// </summary>
+        ///
+        private class ObservableConsumerPartitionReceiver : PartitionReceiver
+        {
+            public string TransportConsumerCreatedWithConsumerGroup { get; private set; }
+            public string TransportConsumerCreatedWithPartitionId { get; private set; }
+            public EventPosition TransportConsumerCreatedWithEventPosition { get; private set; }
+            public EventHubsRetryPolicy TransportConsumerCreatedWithRetryPolicy { get; private set; }
+            public bool TransportConsumerCreatedWithTrackLastEnqueuedEventProperties { get; private set; }
```

Comment:
Congratulations!   You have now graduated to the master level of verbose property naming.   You may now join me in the member's lounge.